src/main/java/me/clip/placeholderapi/expansion/manager/LocalExpansionManager.java

```diff
@@ -323,6 +327,19 @@ public boolean unregister(@NotNull final PlaceholderExpansion expansion) {
     if (expansions.remove(expansion.getIdentifier().toLowerCase(Locale.ROOT)) == null) {
       return false;
     }
+    
+    // Don't unregister expansions that are marked internal or set to be persistent.
+    if (expansion.getExpansionType() == Type.INTERNAL || expansion.persist()) {
+      // Print warning if an external expansion is set to be persistent.
+      if (expansion.getExpansionType() == Type.EXTERNAL && expansion.persist()) {
+        Msg.warn("Nag author(s) %s about their expansion %s being marked as \"external\" " 
```

Comment:
The author didn't mark the expansion type and probably don't know what that means. The message should probably be changed to something more like "Nag author(s) %s about their external expansion having persist() set to true!"

Reply:
+1

Reply:
Adressed this.